### PR TITLE
Improve style encapsulation

### PR DESCRIPTION
**Pull request by lebaudantoine**
## Purpose

Improve DX and code maintainability 

## Proposal

Based on peer feedback, and good practice. Try to encapsulate styling for core components.

### CODE REVIEW

File: src/components/ContentSection.tsx [hunk 1]
```diff
@@ -1,24 +1,31 @@
 import { HTMLAttributes, PropsWithChildren } from 'react'
 import { twMerge } from 'tailwind-merge'
 
-export const ContentSection = ({
+interface ContentSectionProps
+  extends PropsWithChildren,
+    Omit<HTMLAttributes<HTMLDivElement>, 'className'> {
+  backgroundVariant?: 'gray'
```
Comment (manuhabitela) [inline src/components/ContentSection.tsx:7]: why not just `background` instead of `backgroundVariant`?

File: src/components/ContentSection.tsx [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
+  removePadding?: boolean
```
Comment (manuhabitela) [inline src/components/ContentSection.tsx:8]: I'd argue an option that enables padding (which would be true by default) would be easier to reason with. `padding` or `withPadding` I guess. Feels a bit weird to _enable_ an option that _disables_ something.

File: src/components/Footer.tsx [hunk 3]
```diff
@@ -34,9 +34,9 @@ export const Footer = () => (
           <span className="logo-footer-separator" />
           <p className="font-bold leading-5 text-sm xs:text-base">
             Direction
-            <Br className="inline" />
+            <Br showOnMobile />
```
Comment (manuhabitela) [inline src/components/Footer.tsx:37]: al dente :ok_hand:

File: src/pages/404.tsx [hunk 4]
```diff
@@ -5,12 +5,8 @@ import { Button } from '@/components/Button'
 
 export default function NotFound() {
   return (
-    <Layout
-      title="Page introuvable"
-      className="px-4 py-16 md:py-24"
-      layoutClassName="bg-white-1"
-    >
-      <div className="text-center max-w-[26rem] m-auto">
+    <Layout title="Page introuvable">
```
Comment (manuhabitela) [inline src/pages/404.tsx:8]: With this change now the 404 page doesn't have a gray background anymore, is that intentional?

**Review by manuhabitela — COMMENTED** [review]
Cool!

Yeah I find it clearer that way. The `Br` component is a good example why.

I feel it's a good time to get rid of `twMerge` with this logic. It adds a few kilobytes in the client JS bundle just for merging classes, now if we decide to rely on props for styling it's not that useful anymore.

File: src/pages/404.tsx (hunk 4 above)
Comment (lebaudantoine) [reply src/pages/404.tsx:8]: my bad

File: src/components/ExternalLink.tsx [hunk 5]
```diff
@@ -54,7 +53,7 @@ export const ButtonExternalLink: React.FC<LinkProps> = ({
   ...subprops
 }) => (
   <ExternalLink
-    className={twMerge(ButtonStyle, className)}
+    className={`${buttonStyles.default}, ${className || ''}`}
```
Comment (manuhabitela) [inline src/components/ExternalLink.tsx:56]: I think we don't want the comma here!

**Review by manuhabitela — APPROVED** [review]
Great! Just one tiny typo to take care of I think. Thanks!

I guess in the future we could use something like https://github.com/lukeed/clsx to ease up classNames handling but template literals do the trick for sure.

**Pull request merged**